service/storage_proxy.cc

```diff
@@ -689,17 +695,17 @@ class storage_proxy::remote {
         const auto fence = fence_opt.value_or(fencing_token{});
 
         if (auto stale = _sp.apply_fence(fence, src_ip)) {
-            co_return co_await encode_replica_exception_for_rpc<Result>(p->features(), to_future(std::move(*stale)));
+            co_return co_await add_replica_exception_to_query_result<Result>(p->features(), to_future(std::move(*stale)));
```

Comment:
It looks like you could just replace the uses of `add_replica_exception_to_query_result` + `to_future` with just `encode_replica_exception_for_rpc`, no?

Reply:
Good catch. It made the complicated `to_future` redundant.